### PR TITLE
Call Home Enhancements

### PR DESCRIPTION
**Pull request by kobalski**
With this PR hazelcast nodes will send HDMemory configuration of the
cluster and also will send client information connected cluster.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java [hunk 1]
```diff
@@ -1287,6 +1292,59 @@ private void setMembersRef(Map<Address, MemberImpl> memberMap) {
...
+        for (MemberImpl member : node.getClusterService().getMemberList()) {
+            Address target = member.getAddress();
+            InvocationBuilder invocationBuilder = operationService1.createInvocationBuilder(serviceName,
+                    clientInfoOperation, target);
+            Future<Object> future = invocationBuilder.setTryCount(1).invoke();
+            HashMap<String, ClientType> endpoints = new HashMap<String, ClientType>();
+            try {
+                endpoints = (HashMap<String, ClientType>)future.get();
+            } catch (InterruptedException e) {
+                logger.warning("Cannot get client information from : " + target.toString());
```
Comment (mesutcelik) [inline hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java]: so we swallow the exception if we don't get any info from some of the members. @sancar any suggestion here?

File: hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java [hunk 2]
```diff
@@ -154,13 +145,9 @@ private Document fetchWebService(String urlStr) {
...
             conn.setReadTimeout(TIMEOUT * 2);
             in = new BufferedInputStream(conn.getInputStream());
-            final DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            return builder.parse(in);
-        } catch (SAXException ignored) {
-            EmptyStatement.ignore(ignored);
-        } catch (IOException ignored) {
-            EmptyStatement.ignore(ignored);
-        } catch (ParserConfigurationException ignored) {
+            IOUtil.closeResource(in);
```
Comment (mesutcelik) [inline hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java]: that is already called in finally block.

File: hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java [hunk 3]
```diff
@@ -154,13 +145,9 @@ private Document fetchWebService(String urlStr) {
...
+            return null;
```
Comment (mesutcelik) [inline hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java]: I don't think return value is used by anyone so we can change the return type of the method to ```void```

File: hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java [hunk 4]
```diff
@@ -168,4 +155,29 @@ private Document fetchWebService(String urlStr) {
 
         return null;
```
Comment (mesutcelik) [inline hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java]: we can remove this.

File: hazelcast-client-new/src/test/java/com/hazelcast/client/ConnectedClientOperationTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class ConnectedClientOperationTest extends HazelcastTestSupport {
```
Comment (bilalyasar) [inline hazelcast-client-new/src/test/java/com/hazelcast/client/ConnectedClientOperationTest.java]: minor comment, missing category again

File: hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java (hunk 1 above)
Comment (kobalski) [inline hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java]: Increased tryCount to 50.

**Comment by serkan-ozal** [comment]
:+1:

**Pull request merged**